lib/miq_automation_engine/engine/miq_ae_event.rb

```diff
@@ -138,44 +139,9 @@ def self.call_automate(obj, attrs, instance_name, options = {})
     end
   end
 
-  def self.stream_target(object)
-    case object
-    when EmsEvent
-      object.vm_or_template || object.ext_management_system
-    when VmOrTemplate, MiqServer, Service, ExtManagementSystem, MiqRequest
-      object
-    when Storage
-      object.ext_management_systems.first
-    else
-      object.ext_management_system
-    end
-  end
-
-  def self.automate_user_ids(object, event_type)
-    target = stream_target(object)
-    case target
-    when VmOrTemplate, Service
-      group  = target.miq_group
-      user   = target.evm_owner
-      user   = User.super_admin if user.nil? || !user.miq_groups.include?(group)
-      tenant = target.tenant
-    when ExtManagementSystem
-      user   = User.super_admin
-      tenant = target.tenant
-      group  = tenant.default_miq_group
-    when MiqServer
-      user   = User.super_admin
-      tenant = user.current_tenant
-      group  = user.current_group
-    when MiqRequest
-      user   = target.get_user
-      tenant = user.current_tenant
-      group  = user.current_group
-    end
-
-    raise "A group is needed to raise an event. [#{object.class.name}] id:[#{object.id}] event_type: [#{event_type}]" unless group
-    [user.id, group.id, tenant.id]
+  def self.tenant_identity_target(obj)
+    obj.respond_to?(:tenant_identity) ? obj : obj.ext_management_system
   end
 
-  private_class_method :call_automate, :automate_user_ids, :stream_target
+  private_class_method :call_automate, :tenant_identity_target
```

Comment:
I recommend keeping this immediately after each method, instead of lumped together at the end.